src/main/java/io/candydoc/ddd/extract_ddd_concepts/DDDConceptsExtractionService.java

```diff
@@ -64,39 +68,49 @@ private void trackAndApply(List<Event> occurredEvents) {
     occurredEvents.forEach(event -> event.accept(this));
   }
 
-  public void handle(ExtractValueObjects command) {
-    log.info("Extract value objects from {}", command.getPackageToScan());
-    trackAndApply(valueObjectExtractor.extract(command));
-  }
-
   public void handle(ExtractDDDConcepts command) {
     log.info("Extract ddd concepts from {}", command.getPackagesToScan());
-    trackAndApply(boundedContextExtractor.extract(command));
+    List<Event> boundedContextEvents = boundedContextExtractor.extract(command);
+    List<Event> sharedKernelEvents = sharedKernelExtractor.extract(command);
+    if (boundedContextEvents.isEmpty() && sharedKernelEvents.isEmpty()) {
+      throw new NoBoundedContextOrSharedKernelFound(command.getPackagesToScan());
```

Comment:
Ok I get why you removed the bounded context exception.

NoBoundedContext**N**orSharedKernelFound